### PR TITLE
Implement "potential CORS request"

### PR DESCRIPTION
**Pull request by KiChjang**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9516)

### CODE REVIEW

File: components/net_traits/request.rs [hunk 1]
```diff
@@ -144,6 +153,50 @@ impl Request {
         }
     }
 
+    // https://html.spec.whatwg.org/multipage/infrastructure.html#create-a-potential-cors-request
```
Comment (frewsxcv) [inline components/net_traits/request.rs]: drop `infrastructure.html`

**Comment by jdm** [comment]
@nikkisquared Would you like to review this?

**Comment by nikkisquared** [comment]
@jdm sure thing! any pointers for how I should go about it?

**Comment by jdm** [comment]
1. Read the specification that is referenced in the changes.
2. Read the code changes.
3. Ensure that the code changes implements the specification text. Leave comments on lines that are confusing or incorrect, possibly including suggestions for correcting them.
4. Consider if there are parts of the changes that could be written in different ways that would improve clarity. Leave comments with suggestions.
5. Note any stylistic nits that should be addressed.
6. Leave a summary comment to indicate that the review is complete.

**Comment by nikkisquared** [comment]
Thanks @jdm!

**Comment by nikkisquared** [comment]
The destination enum/field doesn't exist, so it can't be set to `subresource` like the potential-CORS request spec asks for. I'm guessing none of the current Fetch implementation uses it, but now might be a good time to put it in.

Other than that, I couldn't find anything substantial needing change. It looks good to me, excepting a minor nit.

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[components/net_traits/request.rs, line 83 \[r2\]](https://reviewable.io:443/reviews/servo/servo/9516#-K9cVkcIj4BLSWkF6cDT:-K9cVkcIj4BLSWkF6cDU:-1494172451) ([raw file](https://github.com/servo/servo/blob/2ff0768db816279e2a40443515bc938ee3804edf/components/net_traits/request.rs#L83)):*
The spec says "The empty string is also a valid keyword, and maps to the Anonymous state." I'm not sure if this helpfully applies to Servo's implementation, or where, but it seems like something to keep in mind.

---

*[components/net_traits/request.rs, line 156 \[r2\]](https://reviewable.io:443/reviews/servo/servo/9516#-K9cTLArjX1hHKR1bhWq:-K9cTLAslOKdvhDFgGfe:495031014) ([raw file](https://github.com/servo/servo/blob/2ff0768db816279e2a40443515bc938ee3804edf/components/net_traits/request.rs#L156)):*
Nit: this should be formatted as a markdown link with `///` instead of `//`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9516#-:-K9cXUXNtG3fUsTZ4JUI:1750104061)*

**Comment by KiChjang** [comment]
@nikkisquared I was planning to create another PR just for updating all the fields for a `Request`, but I'm not sure if it'll conflict with your work.

@bors-servo delegate=nikkisquared

**Comment by nikkisquared** [comment]
@KiChjang Another PR for more `Request` fields would be great, I made a rough list of inconsistencies between the spec and the implementation a while ago, I've put it on gist here: https://gist.github.com/nikkisquared/5817408dcc279500ecac. I don't think adding fields would necessarily conflict with my work, I've had it on the back of my mind and only adding fields as necessary, which is very infrequent.

**Comment by KiChjang** [comment]
@bors-servo retry
* #9528

**Pull request merged**